src/index.rs

```diff
@@ -1133,4 +1106,226 @@ mod tests {
         assert!(new2 > old2);
         assert!(new2 > new1);
     }
+
+    const ACTIONS_COUNT: i32 = 4;
+
+    enum FileAction {
+        CREATE = 0, // create new file by new path
+        UPDATE,     // new content by existing path
+        DELETE,     // remove the file by existing path
+        MOVE,       // same content, new path
+    }
+
+    impl From<i32> for FileAction {
+        fn from(value: i32) -> Self {
+            match value {
+                0 => FileAction::CREATE,
+                1 => FileAction::UPDATE,
+                2 => FileAction::DELETE,
+                3 => FileAction::MOVE,
+                _ => unreachable!("Invalid ActionType value"),
+            }
+        }
+    }
```

Comment:
Maybe it's better to keep all the constants and functions that are not tests at the top of the testing module.